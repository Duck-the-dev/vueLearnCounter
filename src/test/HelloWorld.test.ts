import HelloWorldVue from '../components/HelloWorld.vue'
import { mount } from '@vue/test-utils'

describe('counter', () => {
  test('mount counter', async () => {
    expect(HelloWorldVue).toBeTruthy()

    const wrapper = mount(HelloWorldVue)
    //with props :
    // const wrapper = mount(HelloWorldVue, {
    //   props: {
    //     msg: 'LonesomeT0wn start',
    //   },
    // })

    expect(wrapper.text()).toContain('0')
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('increases counter by one', async () => {
    expect(HelloWorldVue).toBeTruthy()

    const wrapper = mount(HelloWorldVue)

    expect(wrapper.html()).toMatchSnapshot()
    await wrapper.find('#plus').trigger('click')

    expect(wrapper.find('.count').text()).toBe('1')
  })
  test('decreases counter by one', async () => {
    expect(HelloWorldVue).toBeTruthy()

    const wrapper = mount(HelloWorldVue)
    expect(wrapper.html()).toMatchSnapshot()
    await wrapper.find('#minus').trigger('click')

    expect(wrapper.find('.count').text()).toBe('-1')
  })
})
