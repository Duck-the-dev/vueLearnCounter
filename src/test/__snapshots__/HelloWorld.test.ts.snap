// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`counter > decreases counter by one 1`] = `
"<h1 data-v-e17ea971=\\"\\" class=\\"bg-transparent bg-gradient-to-r from-pink-900 to-purple-500 bg-clip-text text-transparent shadow-purple-400 drop-shadow-md duration-200 hover:shadow-xl hover:drop-shadow-2xl\\"></h1>
<div data-v-e17ea971=\\"\\" class=\\"count card border-2 mx-auto mt-5 mb-5 shadow-lg drop-shadow-2xl bg-gray-700 text-center font-bold text-xl\\">0</div>
<div data-v-e17ea971=\\"\\" class=\\"container\\"><button data-v-e17ea971=\\"\\" id=\\"plus\\" class=\\"drawer-button mb-5 mt-5 mx-5 shadow-lg drop-shadow-2xl focus:outline-none text-center font-bold text-xl btn w-1/4 h-1/4\\" type=\\"button\\"> + </button><button data-v-e17ea971=\\"\\" id=\\"minus\\" class=\\"drawer-button mb-5 mt-5 mx-5 shadow-lg drop-shadow-2xl focus:outline-none text-center font-bold text-xl btn w-1/4 h-1/4\\" type=\\"button\\"> - </button></div>
<p data-v-e17ea971=\\"\\"> Edit <code data-v-e17ea971=\\"\\">components/HelloWorld.vue</code> to test HMR </p>
<p data-v-e17ea971=\\"\\"> Check out <a data-v-e17ea971=\\"\\" href=\\"https://vuejs.org/guide/quick-start.html#local\\" target=\\"_blank\\">create-vue</a>, the official Vue + Vite starter </p>
<p data-v-e17ea971=\\"\\"> Install <a data-v-e17ea971=\\"\\" href=\\"https://github.com/johnsoncodehk/volar\\" target=\\"_blank\\">Volar</a> in your IDE for a better DX </p>
<p data-v-e17ea971=\\"\\" class=\\"read-the-docs\\">Click on the Vite and Vue logos to learn more</p>"
`;

exports[`counter > increases counter by one 1`] = `
"<h1 data-v-e17ea971=\\"\\" class=\\"bg-transparent bg-gradient-to-r from-pink-900 to-purple-500 bg-clip-text text-transparent shadow-purple-400 drop-shadow-md duration-200 hover:shadow-xl hover:drop-shadow-2xl\\"></h1>
<div data-v-e17ea971=\\"\\" class=\\"count card border-2 mx-auto mt-5 mb-5 shadow-lg drop-shadow-2xl bg-gray-700 text-center font-bold text-xl\\">0</div>
<div data-v-e17ea971=\\"\\" class=\\"container\\"><button data-v-e17ea971=\\"\\" id=\\"plus\\" class=\\"drawer-button mb-5 mt-5 mx-5 shadow-lg drop-shadow-2xl focus:outline-none text-center font-bold text-xl btn w-1/4 h-1/4\\" type=\\"button\\"> + </button><button data-v-e17ea971=\\"\\" id=\\"minus\\" class=\\"drawer-button mb-5 mt-5 mx-5 shadow-lg drop-shadow-2xl focus:outline-none text-center font-bold text-xl btn w-1/4 h-1/4\\" type=\\"button\\"> - </button></div>
<p data-v-e17ea971=\\"\\"> Edit <code data-v-e17ea971=\\"\\">components/HelloWorld.vue</code> to test HMR </p>
<p data-v-e17ea971=\\"\\"> Check out <a data-v-e17ea971=\\"\\" href=\\"https://vuejs.org/guide/quick-start.html#local\\" target=\\"_blank\\">create-vue</a>, the official Vue + Vite starter </p>
<p data-v-e17ea971=\\"\\"> Install <a data-v-e17ea971=\\"\\" href=\\"https://github.com/johnsoncodehk/volar\\" target=\\"_blank\\">Volar</a> in your IDE for a better DX </p>
<p data-v-e17ea971=\\"\\" class=\\"read-the-docs\\">Click on the Vite and Vue logos to learn more</p>"
`;

exports[`counter > mount counter 1`] = `
"<h1 data-v-e17ea971=\\"\\" class=\\"bg-transparent bg-gradient-to-r from-pink-900 to-purple-500 bg-clip-text text-transparent shadow-purple-400 drop-shadow-md duration-200 hover:shadow-xl hover:drop-shadow-2xl\\"></h1>
<div data-v-e17ea971=\\"\\" class=\\"count card border-2 mx-auto mt-5 mb-5 shadow-lg drop-shadow-2xl bg-gray-700 text-center font-bold text-xl\\">0</div>
<div data-v-e17ea971=\\"\\" class=\\"container\\"><button data-v-e17ea971=\\"\\" id=\\"plus\\" class=\\"drawer-button mb-5 mt-5 mx-5 shadow-lg drop-shadow-2xl focus:outline-none text-center font-bold text-xl btn w-1/4 h-1/4\\" type=\\"button\\"> + </button><button data-v-e17ea971=\\"\\" id=\\"minus\\" class=\\"drawer-button mb-5 mt-5 mx-5 shadow-lg drop-shadow-2xl focus:outline-none text-center font-bold text-xl btn w-1/4 h-1/4\\" type=\\"button\\"> - </button></div>
<p data-v-e17ea971=\\"\\"> Edit <code data-v-e17ea971=\\"\\">components/HelloWorld.vue</code> to test HMR </p>
<p data-v-e17ea971=\\"\\"> Check out <a data-v-e17ea971=\\"\\" href=\\"https://vuejs.org/guide/quick-start.html#local\\" target=\\"_blank\\">create-vue</a>, the official Vue + Vite starter </p>
<p data-v-e17ea971=\\"\\"> Install <a data-v-e17ea971=\\"\\" href=\\"https://github.com/johnsoncodehk/volar\\" target=\\"_blank\\">Volar</a> in your IDE for a better DX </p>
<p data-v-e17ea971=\\"\\" class=\\"read-the-docs\\">Click on the Vite and Vue logos to learn more</p>"
`;
